heat/core/linalg/basics.py

```diff
@@ -827,6 +827,225 @@ def norm(a):
     return exponential.sqrt(d).item()
 
 
+def outer(a, b, out=None, split=None):
+    """
+    Compute the outer product of two 1-D DNDarrays.
+
+    Given two vectors, ``a`` = [a_0, a_1, ..., a_n] and ``b`` = [b_0, b_1, ..., b_m], the outer product is:
+
+    [[a_0*b_0  a_0*b_1 ... a_0*b_m]
+     [a_1*b_0     .
+     [ ...        .
+     [a_n*b_0     .        a_n*b_m]]
+
+    Parameters
+    ----------
+
+    a(n,): DNDarray
+            First input DNDarray. Must be 1-dimensional. Will be flattened by default if more than 1-D
+
+    b(m,): DNDarray
+            Second input DNDarray. Must be 1-dimensional. Will be flattened by default if more than 1-D
+
+    out(n, m): DNDarray, optional
+            A location where the result is stored
+
+    split: int, optional #TODO check out docstring format
+            Split dimension of the resulting DNDarray. Can be 0, 1, or None.
+            This is only relevant if the calculations are memory-distributed,
+            in which case default is ``split=0`` (see Note).
+
+    Note: parallel implementation of outer product, arrays are dense. #TODO sparse #384
```

Comment:
The TODO should probably disappear from the docstring itself, as it would be copied over into the text that is being read by users. Perhaps we could have this in the code?

Reply:
right, fixed